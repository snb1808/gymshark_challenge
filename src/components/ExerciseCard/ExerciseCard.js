import { Component } from 'react';
import Card from 'react-bootstrap/Card';
import './exercise-card.css';
import ExerciseModal from '../ExerciseModal/ExerciseModal';

class ExerciseCard extends Component {
    state = { showInfo: false };

    handleClick = () => this.setState({ showInfo: !this.state.showInfo });

    render = () => {
        const { exercise, gender } = this.props;

        return (
            <Card role='button' className='exercise-card' onClick={this.handleClick}>
                <Card.Img
                    role='button'
                    aria-label={`${exercise.name} image`}
                    variant='top'
                    src={exercise[gender].image}
                />
                <Card.Body>
                    <Card.Title>{exercise.name}</Card.Title>
                     <ExerciseModal content={exercise} show={this.state.showInfo}/>
                </Card.Body>
                <Card.Footer>
                    <small aria-label='body areas' className="text-muted">
                        {exercise.bodyAreas.toString().split(',').join(', ')}
                    </small>
                </Card.Footer>
            </Card>
        )};
}

export default ExerciseCard;