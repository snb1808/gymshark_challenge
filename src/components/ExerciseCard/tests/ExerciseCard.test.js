import { shallow } from 'enzyme';
import ExerciseCard from '../ExerciseCard';

describe('ExerciseCard', () => {
    it('should render correctly', () => {
        const props = { exercise: {
            name: 'Bicep Curls',
                transcript: 'Bicep curls are bicep curls',
                male: { image: 'www.mensimage.com' },
                female: { image: 'www.womensimage.com' },
                bodyAreas: ['Arms', 'Biceps'],
            },
            gender: 'female',
        };

        const component = shallow(<ExerciseCard {...props}/>);

        expect(component.debug()).toMatchSnapshot();
    });
})