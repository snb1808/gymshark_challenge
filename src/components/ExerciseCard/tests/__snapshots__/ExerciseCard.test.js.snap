// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseCard should render correctly 1`] = `
"<Card role=\\"button\\" className=\\"exercise-card\\" onClick={[Function (anonymous)]} body={false}>
  <CardImg role=\\"button\\" aria-label=\\"Bicep Curls image\\" variant=\\"top\\" src=\\"www.womensimage.com\\" />
  <CardBody>
    <CardTitle>
      Bicep Curls
    </CardTitle>
    <ExerciseModal content={{...}} show={false} />
  </CardBody>
  <CardFooter>
    <small aria-label=\\"body areas\\" className=\\"text-muted\\">
      Arms, Biceps
    </small>
  </CardFooter>
</Card>"
`;
